### PR TITLE
Added Message Tests + Minor Changes

### PR DESCRIPTION
**Pull request by MikeVezina**
Not sure why it shows commits that have already been pushed (Maven support + test directory), but it should be fine if we squash + merge.

**Pull Request Main Changes:**
- Added tests for each message object
- Added test suite (with testing utils for Message objects)

**Pull Request Minor Changes:**
- MAX_BLOCK_SIZE is now a public constant in DataMessage
- Changed minimum packet size in RequestMessage to properly conform to TFTP protocol
- Added ability to log byte arrays in a neat & organized fashion

### CODE REVIEW

File: src/main/java/FTPServer.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ public void run() {
 			socket = new DatagramSocket();
 
 			// Parse data into a DAO that is accessible
-			RequestMessage receivedMessage = RequestMessage.parseDataFromPacket(this.packet);
+			RequestMessage receivedMessage = RequestMessage.parseMessageFromPacket(this.packet);
```
Comment (irushavm) [inline src/main/java/FTPServer.java]: Good change. I need to update my branch after this is merged.

File: src/main/java/formats/AckMessage.java [hunk 2]
```diff
@@ -32,7 +32,6 @@ public AckMessage(int blockNum)
 
     /**
      * @return A byte representation of the message
-     * @throws IOException
```
Comment (irushavm) [inline src/main/java/formats/AckMessage.java]: Why take this out?

**Comment by irushavm** [comment]
Need to rebase and fix the method names @MikeVezina

File: src/main/java/logging/Logger.java [hunk 3]
```diff
@@ -90,4 +90,45 @@ private synchronized void log(LogLevel level, String text)
...
+     * @return A comma separated list of all bytes in the array
+     */
+    private synchronized String getByteArrayString(byte[] bytes)
+    {
+        StringBuilder byteStr = new StringBuilder("Byte Array: { ");
+
+        // Add individual bytes to the string
+        for(int i = 0; i < bytes.length; i++) {
+            byte b = bytes[i];
+            byteStr.append(b);
```
Comment (DerekStride) [inline src/main/java/logging/Logger.java]: I like it when the bytes are formatted as 2 digit hex values, what does everyone else think?

```java
String.format("%02x ", b)
```

File: src/main/java/formats/AckMessage.java (hunk 2 above)
Comment (MikeVezina) [reply src/main/java/formats/AckMessage.java]: Must have been done accidentally, will fix!

File: src/test/java/formats/MessageTestSuite.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    static void testInvalidParseData(ParseExecute executor, List<Pair<String, byte[]>> invalidParseData)
+    {
+        System.out.println("Expecting " + invalidParseData.size() + " exceptions.");
+        for(Pair<String, byte[]> data : invalidParseData)
+        {
+            // Expect exceptions to be thrown
+            try {
+                // Parse the data using interface implemented by individual tests
+                executor.parse(data.getValue());
+                fail("Expected InvalidPacketException: " + data.getKey());
```
Comment (DerekStride) [inline src/test/java/formats/MessageTestSuite.java]: This is pretty neat

File: src/main/java/logging/Logger.java [hunk 5]
```diff
@@ -106,4 +106,45 @@ private synchronized void log(LogLevel level, String text)
...
+     * @return A comma separated list of all bytes in the array
+     */
+    private synchronized String getByteArrayString(byte[] bytes)
+    {
+        StringBuilder byteStr = new StringBuilder("Byte Array: { ");
+
+        // Add individual bytes to the string
+        for(int i = 0; i < bytes.length; i++) {
+            byte b = bytes[i];
+            byteStr.append(String.format("0x%02X", b));
```
Comment (DerekStride) [inline src/main/java/logging/Logger.java]: 👍  The `0x` in front of each individual byte might look cluttered but that's not a huge issue.

**Pull request merged**